pkg/object/meshcontroller/meshcontroller.go

```diff
@@ -76,15 +79,18 @@ func (mc *MeshController) DefaultSpec() interface{} {
 // Init initializes MeshController.
 func (mc *MeshController) Init(superSpec *supervisor.Spec, super *supervisor.Supervisor) {
 	mc.superSpec, mc.spec, mc.super = superSpec, superSpec.ObjectSpec().(*spec.Admin), super
+
+	mc.api = api.New(superSpec, super)
+
 	mc.reload()
 }
 
 // Inherit inherits previous generation of MeshController.
-func (mc *MeshController) Inherit(spec *supervisor.Spec,
+func (mc *MeshController) Inherit(superSpec *supervisor.Spec,
 	previousGeneration supervisor.Object, super *supervisor.Supervisor) {
 
-	previousGeneration.Close()
-	mc.Init(spec, super)
+	mc.superSpec, mc.spec, mc.super = superSpec, superSpec.ObjectSpec().(*spec.Admin), super
+	mc.reload()
```

Comment:
`mc.api` is not set as in function `Init`.

Reply:
@xxx7xxxx ， Is that necessary to introduce the `api` into MeshController?  Since `master` and `worker` have handled the API registration inside. And I can't find any use of `m.api` in MeshController.

Reply:
It's expected. Otherwise it will cause repeating registering apis.

Reply:
if so, in `Init`, can we change
```go
mc.api = api.New(superSpec, super)
```
to
```go
api.New(superSpec, super)
```
and remove `mc.api`?

it is weird to only initialize the variable in `Init`.

Reply:
I don't see the difference. IMO, the standalone `api.New` is weirder to me. I updated it to `api.Register`